nbgrader/exchange/exchange.py

```diff
@@ -121,12 +133,28 @@ def copy_files(self):
     def do_copy(self, src, dest):
         """Copy the src dir to the dest dir omitting the self.coursedir.ignore globs."""
         shutil.copytree(src, dest, ignore=shutil.ignore_patterns(*self.coursedir.ignore))
+        # copytree copies access mode too - so we must add go+rw back to it if
+        # we are in groupshared.
+        if self.groupshared:
+            for dirname, _, filenames in os.walk(dest):
+                # dirs become ug+rwx
+                try:   os.chmod(dirname, (os.stat(dirname).st_mode|0o2770) & 0o2777)
+                except PermissionError: pass
+                for filename in filenames:
+                    filename = os.path.join(dirname, filename)
+                    try:    os.chmod(filename, (os.stat(filename).st_mode|0o660) & 0o777)
+                    except PermissionError: pass
+
 
     def start(self):
         if sys.platform == 'win32':
             self.fail("Sorry, the exchange is not available on Windows.")
 
-        self.ensure_root()
+        if not self.groupshared:
```

Comment:
Hmm, I think this check is mainly to ensure that the current user can access the exchange, so I think it is probably still worthwhile to keep this check in even when self.groupshared is set. Unless there's some reason why this check doesn't work that I'm missing?

Reply:
My commit message was `groupshared: don't force exchange root +w for everyone`.  I can't seem to remember the exact reason, but... maybe it's because now that instructors have a special group, we don't need to be g+w for everyone.  But this doesn't necessarily seem specific to groupshared.

One other thing I remembered was tring to reset permissions or force-create the directory.  But, users can't chmod dirs owned by others, so it fails.  This may be realated to me not setting others+write on the top-level exchange directory.

I should think about this more and see if it's really needed.  I probably have to remove it and see what breaks!

Reply:
OK, this is the same comment as below (https://github.com/jupyter/nbgrader/pull/1000#discussion_r305368967).  Discussion about it is written there.  I'll leave this sub-conversation open but do discussion down there.